api/types/network/network.go

```diff
@@ -18,6 +19,7 @@ type IPAMConfig struct {
 	IPRange    string            `json:",omitempty"`
 	Gateway    string            `json:",omitempty"`
 	AuxAddress map[string]string `json:"AuxiliaryAddresses,omitempty"`
+	Options    map[string]string `json:"Options,omitempty"`
```

Comment:
Please remove this from here. As discussed, from docker UX perspective this is not a subnet specific option and hence should not be part of IPAMConfig struct. libnetwork must do the appropriate work to pass any options to the ipam driver during the request calls.